src/server/system_services/bucket_server.js

```diff
@@ -1977,6 +1977,25 @@ function validate_replication(req) {
         pref_by_dst_bucket[destination_bucket] = pref_by_dst_bucket[destination_bucket] || [];
         pref_by_dst_bucket[destination_bucket].push((filter && filter.prefix) || '');
     }
+
+    if (req.rpc_params.replication_policy.log_replication_info) {
+        const logs_bucket = req.system.buckets_by_name &&
+            req.system.buckets_by_name[req.rpc_params.replication_policy.log_replication_info.logs_location.logs_bucket.unwrap()];
+
+        // validate logs_bucket exist
+        if (!logs_bucket) {
+            throw new RpcError('INVALID_LOG_REPLICATION_INFO', `logs bucket is invalid`);
+        }
+
+        const logs_bucket_info = get_bucket_info({ bucket: logs_bucket });
+        if (logs_bucket_info.namespace && logs_bucket.namespace.write_resource.resource.connection.endpoint_type !== 'AWS') {
+            throw new RpcError('INVALID_LOG_REPLICATION_INFO', `logs bucket is not compatible with log based replication`);
```

Comment:
we need to add here to the error message that logs_bucket should be a namespace bucket on AWS